stdgo/reflect/Reflect.hx

```diff
@@ -25,63 +25,14 @@ enum GT_enum {
 	GT_string;
 	GT_unsafePointer;
 	GT_chan(elem:GT_enum);
-	GT_interface;
+	GT_interface(name:String,type:GT_enum);
 	GT_ptr(elem:GT_enum);
 	GT_slice(elem:GT_enum);
 	GT_array(elem:GT_enum, len:Int);
 	GT_func(input:Array<GT_enum>, output:Array<GT_enum>);
 	GT_map(key:GT_enum, value:GT_enum);
-	GT_struct(fields:Array<StructField>);
-	// go2hx Type system internal enum, no Kind equivalent
-	GT_namedType;
-}
-
-
-// TODO - make a go2hx standard class, so that it can be accurately reflected upon
-typedef Method = {
-	// Name is the method name.
-	name:GoString,
-	// PkgPath is the package path that qualifies a lower case (unexported)
-	// method name. It is empty for upper case (exported) method names.
-	// The combination of PkgPath and Name uniquely identifies a method
-	// in a method set.
-	// See https://golang.org/ref/spec#Uniqueness_of_identifiers
-	pkgPath:GoString,
-	type:Type,
-	// func with receiver as first argument
-	func:Value,
-	// index for Type.Method
-	index:GoInt
-}
-
-typedef MethodInfo = {
-	m:Method,
-	haxeName:GoString
-}
-
-typedef NamedTypeData = {
-	typeName:GoString,
-	haxeTypeName:GoString,
-	packPath:GoString,
-	methods:Array<MethodInfo>,
-	underlying:Type,
-	isInterface:Bool,
-	interfaces:Array<GoString>
-};
-
-class StructField implements StructType {
-	public var name:GoString;
-	public var pkgPath:GoString;
-	public var type:Type;
-	public var tag:StructTag;
-	public var offset:GoUIntptr;
-	public var index:Slice<GoInt>;
-	public var anonymous:Bool;
-	public var _address_:Int;
-	public function new(?name:GoString,?pkgPath:GoString,?type:Type,?tag:StructTag,?offset:GoUIntptr,?index:Slice<GoInt>,?anonymous:Bool) {
-		stdgo.internal.Macro.initLocals();
-		_address_ = ++Go.addressIndex;
-	}
+	GT_struct(fields:Array<GT_enum>);
```

Comment:
How will the name and `tag` of the field be passed through to `reflect.Type.field()` so that it can respond with a  completed `StructField` for that field?

Reply:
I see from a comment that `GT_namedType` might be used for fields:
```
GT_namedType(name:String,type:GT_enum,interfaces:Array<GT_enum>); //used for fields and named structs/interfaces
```
Which would certainly solve the name issue, but not the https://golang.org/pkg/reflect/#StructTag issue.